src/controller/c/robot.c

```diff
@@ -159,6 +159,8 @@ void stream_pipe_read(int fd, char **buffer) {
     len = 0;
 #endif
   (*buffer)[len] = '\0';
+  if (!len)
+    *buffer = NULL;
```

Comment:
That's a memory leak: the pointer to the allocated 1024 bytes buffer is lost, so the memory cannot be free.
```suggestion
  if (len != 0)
    (*buffer)[len] = '\0';
  else {
    free(*buffer);
    *buffer = NULL;
  }
```